ChessGame/static/chessgame/js/ChessBoard/chessSquare.js

```diff
@@ -1,17 +1,40 @@
 class ChessSquare {
   highlighted = false;
+  chessPiece = new ChessPiece();
+  position;
+  chessSquare;
+  color;
+
+  constructor(position, color) {
+    this.position = position;
+    this.color = color;
+  }
 
   highlight = () => {
-    this.highlighted = !this.highlighted;
+    this.highlighted = true;
+  };
+
+  unhighlight = () => {
+    this.highlighted = false;
   };
 
-  display = (color) => {
-    const chessSquare = document.createElement('div');
-    chessSquare.className = 'square';
-    chessSquare.style.backgroundColor = color;
+  setChessPiece = (chessPiece) => {
+    this.chessPiece = chessPiece;
+  };
+
+  display = () => {
```

Comment:
This is an interesting approach! Is the idea to call display on each square / piece once during a game or what?

Reply:
To rerender all the squares every turn to update all the pieces

Reply:
Makes sense!